django/utils/translation/__init__.py

```diff
@@ -23,6 +29,39 @@
 LANGUAGE_SESSION_KEY = '_language'
 
 
+@receiver(autoreload_started, dispatch_uid='translation_autoreload_started')
+def autoreload_started(sender, **kwargs):
+    from django.conf import settings
+    from django.conf import locale
+
+    if settings.USE_I18N:
+        locale_dir = Path(locale.__file__).parent
+        directories = [
+            locale_dir,
+            Path('locale'),
+            *settings.LOCALE_PATHS
+        ]
+        directories.extend(Path(config.path) / 'locale' for config in apps.get_app_configs())
+        directories.extend(Path(p) for p in settings.LOCALE_PATHS)
+
+        for path in directories:
+            absolute_path = path.absolute()
+            sender.watch(absolute_path, '**/*.mo')
+
+
+@receiver(file_changed, dispatch_uid='translation_file_changed')
+def translation_file_changed(sender, file_path, **kwargs):
+    if file_path.suffix == '.mo':
```

Comment:
It's possible we need to guard this code with `if settings.USE_I18N` is some way.

Also, I'd like to make sure we are not interfering with the `Trans/_trans`  black magic this module performs by putting this signal receiver here.

Reply:
Can you elaborate on the black magic? Perhaps this should be guarded, I'll add that

Reply:
> Can you elaborate on the black magic?

Currently the the `django.utils.translation` API functions are set to either the ones in `trans_real.py` (which use gettext) or to the ones in `trans_null.py` (no ops) based on the value of `USE_I18N`,  through the use of some clever code:
a) The `USE_I18N` evaluation is performed once per process (interpreter?) and function b) In a way in which the calls are actually short circuited instead of being called from a stub function c) The code in charge of this removes itself from the scene afterwards. 

That magic is performed here in this `__init__.py` file. AFAIK this is done to reduce the run-time impact of the i18n machinery in the `USE_I18N = False` case.

But I've tested this PR since I wrote the above comment and the changes you propose doesn't seem to disrupt it.